### ISSUE TITLE
logits cause error if they exceed num_classes

### ISSUE THREAD

**Issue by cemde** [PyTorchLightning/metrics#358]
## 🐛 Bug

When passing logits as predictions into some metrics, torchmetrics raises an error "The highest label in `preds` should be smaller than `num_classes`."

The docs for F1 state:
```
Accepts logits or probabilities from a model output or integer class values in prediction. Works with multi-dimensional preds and target
```

This error message is due to the `preds.max() >= num_classes` (`utilities/checks.py`, line 166) 

However, the logits CAN be larger than num_classes. For instance, logits of `[1,6,1,1]` become softmax of `[0.0066, 0.98, 0.0066, 0.0066]`, which is valid.

### To Reproduce

Steps to reproduce the behaviour:

```
from torchmetrics.utilities.checks import _check_num_classes_mc
_check_num_classes_mc(preds=torch.tensor([1,10,1,1]), target=torch.tensor([1,0,0,0]), num_classes=4, multiclass=None, implied_classes=4)
```

#### Code sample

see above

### Expected behavior

calculate the metric properly

### Environment

 - TorchMetrics 0.4.1
 - PyTorch Version (e.g., 1.0): 1.9.0
 - OS (e.g., Linux): MacOS 11.4
 - How you installed PyTorch (`conda`, `pip`, source): conda -c pytorch
 - Build command you used (if compiling from source):
 - Python version: 3.9.2
 - CUDA/cuDNN version: CPU
 - GPU models and configuration: CPU
 - Any other relevant information: 

### Additional context

seems like the same issue as

https://github.com/PyTorchLightning/metrics/issues/354

**Comment by Borda** [comment]
@cemde mind check fix in #357?
